docs/criusupport.md

```diff
@@ -0,0 +1,99 @@
+
+Because the checkpoint image files have the live state of the VM that is used to restore the VM in different systems, they should not contain cryptographically-sensitive security data. If the image files contain sensitive security data, the security components are vulnerable to exploitation even if you don't move the image files between systems. The CRIU technical preview introduces the `CRIUSECProvider` security provider, which provides the following limited set of security services: `MessageDigest` and `SecureRandom`. When you enable CRIU support, all existing security providers are removed from the security provider list during the checkpoint phase and `CRIUSECProvider` is added. When you restore the VM in the nonportable restore mode ([`-XX:+CRIURestoreNonPortableMode`](xxcriurestorenonportablemode.md)), `CRIUSECProvider` is removed from the security provider list and the previous security providers are added back again.
+
+When the VM starts from the saved point instead of the beginning, the VM startup time improves and hence results in better VM performance.
+
+## Enabling CRIU support
+
+CRIU support is not enabled by default. You must enable it by specifying the [`-XX:+EnableCRIUSupport`](xxenablecriusupport.md) command-line option when you start your application.
+
+## Configuring CRIU support
+
+You can access the OpenJ9 CRIU support capabilities by specifying different options. The VM enables the addition of VM options on restore through the `CRIUSupport.registerRestoreOptionsFile` API and environment variables through the `CRIUSupport.registerRestoreEnvVariables` API. `OPENJ9_RESTORE_JAVA_OPTIONS` is a special environment variable for adding the VM options on restore.
+
+There are new options that work at checkpoint as well as at restore and some new options that work at restore only. There are also some existing options that work on restore but some of them behave differently.
+
+### New options
+
+You can use the following options only during the checkpoint phase:
+
+- [`-XX:[+|-]CRIURestoreNonPortableMode`](xxcriurestorenonportablemode.md): Specifies whether the JIT and AOT compiler generates nonportable compiled code on restore.
+- [`-XX:CheckpointGCThreads`](xxcheckpointgcthread.md): Reduces the number of garbage collection (GC) threads that exist when you create a checkpoint image, to speed up the later restoration of the checkpoint image.
```

Comment:
`XX:CheckpointGCThreads` state by default and when CRIU is enabled may need to be spelled out more clearly ? In fact maybe this is needed even for the `[`-XX:[+|-]CRIURestoreNonPortableMode`]` option ?

Reply:
Yes. Links are provided to move to the corresponding topics for more information.